angler: support Wi-Fi and graphics

zhuowei opened the pull request:
With these patches, the Nexus 6P supports Wi-Fi and ~almost~ actually supports Weston.

Issues/questions:

- ~Graphics still don't work properly: I had to run `while [ ! ]; do cat /sys/class/graphics/fb0/modes > /sys/class/graphics/fb0/mode; done`. Not sure [how the Nexus 5 fixed this](https://github.com/postmarketOS/pmbootstrap/pull/66); I guess I can add a patch in Weston to automatically run this (or the equivalent ioctl) every frame. (With that command running, graphics work well.)~ msm-fb-refresher fixes this
- ~(Note that the Android recovery does send this ioctl every frame for double buffering)~
- ~KGSL is disabled in the kernel, so freedreno won't work, and everything's software rendered.~ KGSL is not disabled in the kernel anymore, although I haven't enabled the KGSL_DRM compat layer in the kernel yet, nor have I packaged the a440 blobs, so freedreno still won't run. That's for later.
- ~Should a kernel have NFS enabled? I had it enabled for testing.~ - based on review, yes.
- ~Wi-Fi uses the blobs from linux-firmware, which are old; I can package the AOSP blobs (used on this device's stock firmware) that are more up to date if you want. (see the discussion on #513)~ newer blobs packaged.
- The Wi-Fi calibration file is different for various revisions of this hardware; I've added the one that was for my device, but it may not work for earlier/later revisions. Android chooses the correct calibration file at boot in the init.angler.rc.

Tested over an NFS mounted root, but it should work on regular devices.

zhuowei commented:
@ollieparanoid I ended up making a Weston patch (#558) as I'm not sure how to modify the kernel. Do you have any advice on fixing the kernel instead?

Re. AOSP blobs: should I package all of them or just the bcm4358 needed for angler?

ollieparanoid commented:
I think the weston patch is a good solution, if I understand correctly the double-buffering isn't a bug, but valid usage of the framebuffer API.
In my opinion, it is enough to package just the blob needed for angler.

drebrez commented on aports/device/device-huawei-angler/90-android.rules:
```diff
@@ -1,2 +1,7 @@
 # udev rules file
 # All device names can be read from weston's logfile (/tmp/weston.log in postmarketOS)
+# Not calibrated for now.
```
@zhuowei is the "Not calibrated for now" comment still valid?

zhuowei replied:
It seems that the default `1 0 0 0 1 0` calibration works the best, actually; I'll switch it over

drebrez commented on aports/device/device-huawei-angler/deviceinfo:
```diff
@@ -29,4 +29,4 @@ deviceinfo_flash_offset_tags="0x01e00000"
 deviceinfo_flash_pagesize="4096"
 
-deviceinfo_weston_core_modules="xwayland.so"
+deviceinfo_weston_core_modules=""
```
@zhuowei you removed the "xwayland.so" by purpose? It doesn't work for this device?

zhuowei replied:
It does work; I removed it for debugging; I'll add it back

zhuowei commented:
Rebased, squashed into two commits, updated to latest AOSP Wi-Fi firmware blobs, enabled msm-fb-refresher.

With these changes, Nexus 6P supports screen output and Wi-Fi (yay!)

This is ready for review and merge.

Note that I only tested this with NFS: I'll probably try flashing the built image later. Some newer devices require a sparse image for flashing, but I haven't tested this for Nexus 6P yet.

zhuowei commented on aports/device/device-huawei-angler/90-android-touch-dev.rules:
```diff
@@ -1,2 +1,7 @@
+SUBSYSTEM=="input", ATTRS{name}=="synaptics_dsx", \
+ENV{ID_INPUT}="1", ENV{ID_INPUT_TOUCHSCREEN}="1" \
+ENV{ID_INPUT_TOUCHPAD}="" ENV{ID_INPUT_KEY}="" \
+ENV{LIBINPUT_CALIBRATION_MATRIX}="1 0 0 0 1 0"
```
it's to explicitly indicate that we are using default config; it's not actually needed.

ollieparanoid replied:
This is fine for me. we could discuss in #182 whether we prefer the explicit notation or leaving it out and then apply it to all devices at once.

ollieparanoid commented:
In my opinion, this PR is ready to be shipped! :ship: Would it be OK for you if we squashed the two commits into one? That is how we usually do it. If not, here are some options: https://github.com/postmarketOS/pmbootstrap/pull/481#issuecomment-328563246.

zhuowei commented:
@ollieparanoid of course. I had a separate commit for the firmware to make review easier; feel free to squash into one.
